### ISSUE TITLE
speed zero

### ISSUE THREAD

**Issue by langxj** [traccar/traccar-client-android#334]
Hello, I use this APP on my smartphone, the server is deployed on my cloud platform, the program can run correctly, but the speed is always zero (the speed of the test is 50Km/h).

**Comment by tananaev** [comment]
Have you changed accuracy to HIGH?

**Comment by langxj** [comment]
I don't change it. I'll try it. Thank you for your enthusiasm

**Comment by langxj** [comment]
@tananaev Hello master.I set up according to your prompt, but the speed is still 0.

**Comment by tananaev** [comment]
OK, are you sure that you have a GPS signal?

**Comment by langxj** [comment]
,Yes,i am sure. I've changed three mobile phones, speed still 0。

**Comment by tananaev** [comment]
OK, how do you know that it has?

**Comment by langxj** [comment]
My car is on the freeway, I use GPS Status APP,Can get GPS

**Comment by langxj** [comment]
![_20180603174306](https://user-images.githubusercontent.com/7477494/40885284-17be6054-6756-11e8-8f48-5d24f5d58056.png)

**Comment by tananaev** [comment]
Both course and speed are zero, so it's most likely network location, not GPS.

**Comment by langxj** [comment]
@tananaev Hi master.The report screenshot : 
![image](https://user-images.githubusercontent.com/7477494/40885363-70ce9974-6757-11e8-8c31-6046f13dfc10.png)

**Comment by langxj** [comment]
@tananaev My GPS has been opened, how to set up the GPS update location.

**Comment by tananaev** [comment]
OK, please provide screenshots:

1. Android settings showing that GPS is enabled.
2. App settings screen showing that high accuracy is selected.
3. Screenshot with GPS icon on the status bar showing that you have a fix.

**Comment by langxj** [comment]
It's dark. I'll test it tomorrow. Thank you very much.

**Comment by langxj** [comment]
@tananaev The problem is solved. The location mode of mobile phone is set to use GPS only。Thank you for making such a great procedure.

**Comment by ovizii** [comment]
I am a new user of tracar and also had the issue of no speed showing for my android client. Did I understand correctly that for speed to show I need to set the accuracy on the client side to high? 

is there a documentation for the other parameters, i.e. Frequency, Distance and Angle?

**Comment by oliv3** [comment]
Speed needs GPS to be turned on

**Comment by tananaev** [comment]
Yes, accuracy needs to be set to high.

**Comment by oliv3** [comment]
Fused location can get a speed without GPS ?

**Comment by tananaev** [comment]
It might get it from other sensors (e.g. magnetometer).

**Comment by oliv3** [comment]
Ok, my bad. Read that somewhere but never got this case, please ignore my answer :)
So it's really better, then. Cool !

**Comment by ovizii** [comment]
Thanks for confirming. I set accuracy to high and I got speed information when driving home.

**Comment by stefanb** [comment]
It seems #352 would fix this problem - send null (or not send it at all), show " " or "N/A" instead of 0 when there is no data so that user isn't fooled that it is zero.

**Comment by oliv3** [comment]
@stefanfb To be honest, #352 will fix part of the problem, because server-side unknown or undefined values will default to 0, and not NULL:

 * latitude/longitude
 * speed
 * bearing
 * altitude
 * accuracy

#352 is a small step towards having the problem solved, but upstream doesn't seem like wanting to fix database schema issues.

**Comment by oliv3** [comment]
https://github.com/traccar/traccar-client-android/pull/353

**Comment by stefanb** [comment]
Yes, all those optional values should be nullable in the DB as well, otherwise the DB will contain some fake default values, most likely zeros.

**Comment by oliv3** [comment]
@stefanfb There are corresponding (closed) issues in the server repository.
